### ISSUE TITLE
Having multiple GUI selected and moving them can make them move at different amounts

### ISSUE THREAD

**Issue by subsoap** [defold/editor2-issues#1685]
![bug](https://user-images.githubusercontent.com/409170/36449547-ee08f18c-163f-11e8-9031-9a5e1d557e20.gif)

https://gfycat.com/GlaringDismalEft

**Comment by matgis-king** [comment]
It is kind of hard to tell from the gif, but it sort of looks like you're translating `xp_clipper` and its child `xp_fg` at the same time by having them both selected. That would offset `xp_fg` in relation to its parent also.

Maybe the unexpected behavior here is actually that `xp_fg` was selected along with its parent during a box selection?

**Comment by Jerakin** [comment]
Yeah, some would say this is the expected behaviour. This is how it works in many 3D softwares.
Selecting both the parent and the child and moving them both 10px would make the child move 20px because it is parented to something that also moves 10px and not only the child.

**Comment by subsoap** [comment]
That makes sense!

**Comment by subsoap** [comment]
@matgis-king I think the issue I'm really having with this is that when I draw and select a bunch of GUI elements I cannot transform them. What should happen when click and drag rect select it should only select the top level node of each selection maybe? I'll reopen because I guarantee this will be a reoccurring issue with designers down the road.

![guiz](https://user-images.githubusercontent.com/409170/36517048-9d2256d6-1735-11e8-9410-187b1043cc19.gif)

**Comment by subsoap** [comment]
To add, if I ctrl click multiple parent nodes in the Outline the selection is correct. Box select selection is incorrect.

**Comment by matgis-king** [comment]
Yes I think your analysis is correct. Box selection should probably select the same thing as individual clicks would have selected.

**Comment by AGulev** [comment]
related? https://github.com/defold/editor2-issues/issues/1121
